### PR TITLE
Adding VGo and separating non-official package managers

### PR DESCRIPTION
**Pull request by tobbbles**
Please check if what you want to add to `awesome-go` list meets [quality standards](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#quality-standard) before sending pull request. Thanks!

**Please provide package links to:**

- github.com repo: https://github.com/golang/vgo
- godoc.org: https://godoc.org/golang.org/x/vgo
- goreportcard.com: https://goreportcard.com/report/golang.org/x/vgo
- coverage service link ([cover.run](https://cover.run/), [gocover](http://gocover.io/), [coveralls](https://coveralls.io/) etc.), example: `cover.run go`

Very good coverage

**Note**: that new categories can be added only when there are 3 packages or more.

**Make sure that you've checked the boxes below before you submit PR:**
- [X] I have added my package in alphabetical order.
- [X] I have an appropriate description with correct grammar.
- [X] I know that this package was not listed before.
- [X] I have added godoc link to the repo and to my pull request.
- [ ] I have added coverage service link to the repo and to my pull request.
- [X] I have added goreportcard link to the repo and to my pull request.
- [X] I have read [Contribution guidelines](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#quality-standard).

Thanks for your PR, you're awesome! :+1:

### CODE REVIEW

**Comment by tobbbles** [comment]
Note that I kept `dep` as it's [still suggested to be used](https://github.com/golang/go/wiki/vgo#current-state) until vgo is fully released.

**Comment by kirillDanshin** [comment]
@tobbbles shout-out for your contribution. please revert removing all non-deprecated and supported managers. we just should add notes that those are not official solutions

**Comment by tobbbles** [comment]
@kirillDanshin Done as you asked, I also reworded the official package management tools to better reflect their purpose

File: README.md [hunk 1]
```diff
@@ -1016,9 +1016,13 @@ See [go-hardware](https://github.com/rakyll/go-hardware) for a comprehensive lis
 
 ## Package Management
 
-*Libraries for package and dependency management.*
+*Official tooling for package management*
 
 * [dep](https://github.com/golang/dep) - Go dependency tool.
+* [vgo](https://go.googlesource.com/vgo/) - Versioned Go.
```
Comment (kirillDanshin) [inline README.md]: that's 404 :)

Comment (tobbbles) [reply README.md]: Interesting, it does not 404 for me, see below
image

Comment (nicklanng) [reply README.md]: Nor me!

Comment (kirillDanshin) [reply README.md]: oh, never mind, got fixed for me

**Pull request merged**